clients/src/main/java/org/apache/kafka/common/record/MemoryRecords.java

```diff
@@ -56,8 +56,14 @@ public int sizeInBytes() {
 
     @Override
     public long writeTo(GatheringByteChannel channel, long position, int length) throws IOException {
-        ByteBuffer dup = buffer.duplicate();
+        if (position > Integer.MAX_VALUE)
```

Comment:
This is a good question. Would you mind if I look into that as a follow-up? I need to check if `TransportLayer` should be changed too.